src/SSTemplateParser.peg

```diff
@@ -1309,8 +1309,8 @@ EOC;
     {
         // If this template contains a doctype, put it right after it,
         // if not, put it after the <html> tag to avoid IE glitches
-        if (stripos($code ?? '', "<!doctype") !== false) {
-            $code = preg_replace('/(<!doctype[^>]*("[^"]")*[^>]*>)/im', "$1\r\n<!-- template $templateName -->", $code ?? '');
+        if (preg_match('/(?<!preg_match\(\'\/)<!doctype/i', $code)) {
```

Comment:
Is this right? The string `preg_match` is here twice, once as a function call and once as part of the regex

If this is correct, please an inline comment to at least say that it's intentional

Reply:
Yes, we're using `preg_match` to look for a match, and the match we're looking for must not be prepended with `preg_match(`. I'll add a comment about it